### PR TITLE
backend: github: keep track of rate limit

### PR DESCRIPTION
**Pull request by smonero**
### Description

Use the header response to keep track of the github rate limit

### Testing Performed

make backend-test

### GitHub Issue

Fixes #

### CODE REVIEW

**Comment by mikecutalo** [comment]
Will want to set this gauge in OSS for any and all GH request, we dont need to pass it back to the caller if that helps.

File: backend/service/github/github.go [hunk 1]
```diff
@@ -95,7 +95,10 @@ func (s *svc) CreateIssueComment(ctx context.Context, ref *RemoteRef, number int
 	com := &githubv3.IssueComment{
 		Body: strPtr(body),
 	}
-	_, _, err := s.rest.Issues.CreateComment(ctx, ref.RepoOwner, ref.RepoName, number, com)
+	_, response, err := s.rest.Issues.CreateComment(ctx, ref.RepoOwner, ref.RepoName, number, com)
+	if response != nil {
+		s.scope.SubScope("github_client").Gauge("rate_limit_remaining").Update(float64(response.Rate.Remaining))
```
Comment (danielhochman) [inline backend/service/github/github.go:100]: looks good but move to a method invocation e.g.
```go
emitRateLimit(response)
```

i'm also going to look to see if it's possible to just hook the http client so every method doesn't need to remember to call that

**Comment by danielhochman** [comment]
Ok something like this should work...


```go
type StatsRoundTripper struct {
	Wrapped http.RoundTripper
	scope tally.Scope
}

func (st *StatsRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
	resp, err := st.Wrapped.RoundTrip(req)

	if hdr := resp.Header.Get("X-RateLimit-Remaining"); hdr != "" {
		if v, err := strconv.Atoi(hdr); err == nil {
			st.scope.Gauge("foo").Update(float64(v))
		}
	}
	return resp, err
}

func newService() {
...
    httpClient.Transport = &StatsRoundTripper{Wrapped: httpClient.Transport}
    restClient := githubv3.NewClient(httpClient)
...
}


```

File: backend/service/github/github.go [hunk 2]
```diff
@@ -40,12 +41,28 @@ const CurrentUser = ""
...
+	Wrapped http.RoundTripper
+	scope   tally.Scope
+}
+
+func (st *StatsRoundTripper) RoundTrip(req *http.Request) (*http.Response, error) {
+	resp, err := st.Wrapped.RoundTrip(req)
+
+	if hdr := resp.Header.Get("X-RateLimit-Remaining"); hdr != "" {
+		if v, err := strconv.Atoi(hdr); err == nil {
+			st.scope.Gauge("github_rate_limit_remaining").Update(float64(v))
```
Comment (mikecutalo) [inline backend/service/github/github.go:54]: whats the full stats namespace? I think its going to be `clutch.service.github.github_rate_limit_remaining` ? If so lets drop the extra github, you should be able to test this locally.

**Review by danielhochman — APPROVED** [review]

**Review by mikecutalo — APPROVED** [review]

**Pull request merged**